specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json

```diff
@@ -0,0 +1,1222 @@
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The workspace for Grafana resource is deleted."
+          },
+          "204": {
+            "description": "NoContent"
+          },
+          "default": {
+            "description": "Common error response for all Azure Resource Manager APIs to return error details for failed operations.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "AzureBareMetalInstancesListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AzureBareMetalInstance"
+          },
+          "description": "The list of Azure Bare Metal Instances."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of Azure Bare Metal Instances."
+        }
+      },
+      "description": "The response from the List Azure Bare Metal Instances operation."
+    },
+    "AzureBareMetalInstance": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/AzureBareMetalInstanceProperties",
+          "description": "AzureBareMetal instance properties"
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/systemData",
+          "description": "The system metadata relating to this resource."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "description": "AzureBareMetal instance info on Azure (ARM properties and AzureBareMetal properties)"
+    },
+    "AzureBareMetalInstanceProperties": {
+      "type": "object",
+      "properties": {
+        "hardwareProfile": {
+          "$ref": "#/definitions/HardwareProfile",
+          "description": "Specifies the hardware settings for the Azure Bare Metal Instance."
+        },
+        "storageProfile": {
+          "$ref": "#/definitions/StorageProfile",
+          "description": "Specifies the storage settings for the Azure Bare Metal Instance disks."
+        },
+        "osProfile": {
+          "$ref": "#/definitions/OSProfile",
+          "description": "Specifies the operating system settings for the Azure Bare Metal Instance."
+        },
+        "networkProfile": {
+          "$ref": "#/definitions/NetworkProfile",
+          "description": "Specifies the network settings for the Azure Bare Metal Instance."
+        },
+        "azureBareMetalInstanceId": {
```

Comment:
Is it an azure resource ID? If so, please add "format": "arm-id" and "x-ms-arm-id-details".

https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details

Reply:
azureBareMetalInstanceId property is a unique uuid. We don't use it in any API as parameter. It's just a unique identifier that allow us to check uniqueness between assigned server and customer.